src/System.Diagnostics.Process/tests/ProcessStreamReadTests.cs

```diff
@@ -88,6 +88,243 @@ public void TestAsyncOutputStream()
+            Assert.True(childLockList["childProduced789"].TryAcquire(WaitInMS), "Child process should acquire 'childProduced789' lock");
+
+            // Shutdown process
+            Assert.True(childLockList["shutdownChildProcess"].WaitSignal(WaitInMS), "Signal to shutdown not come after 'WaitInMS'");
+
+            return SuccessExitCode;
+        }
+
+        private Dictionary<string, FileSystemLock> TestAsyncOutputStream_BeginCancelBegin_OutputReadLine_LockHelper(string directoryName)
+        {
+            return new Dictionary<string, FileSystemLock>
+            {
+                { "childStarted", new FileSystemLock("childStarted", directoryName) },
+                { "parentFirstBeginOutputReadLine", new FileSystemLock("parentFirstBeginOutputReadLine", directoryName) },
+                { "parentSecondBeginOutputReadLine", new FileSystemLock("parentSecondBeginOutputReadLine", directoryName) },
+                { "parentCancelOutputRead", new FileSystemLock("parentCancelOutputRead", directoryName)},
+                { "childProduced123", new FileSystemLock("childProduced123", directoryName) },
+                { "childProduced456", new FileSystemLock("childProduced456", directoryName) },
+                { "childProduced789", new FileSystemLock("childProduced789", directoryName) },
+                { "shutdownChildProcess", new FileSystemLock("shutdownChildProcess", directoryName) }
+            };
+        }
+
+        /// <summary>
+        /// We save lock files in temp directory
+        /// </summary>
+        private class FileSystemLock : IDisposable
+        {
+            private readonly string _lockFileName = "";
+            private FileStream _fs = null;
+
+            public FileSystemLock(string lockName, string directoryName)
+            {
+                string rootDirectory = Path.Combine(Path.GetTempPath(), directoryName);
+
+                if (!Directory.Exists(rootDirectory))
+                {
+                    Directory.CreateDirectory(rootDirectory);
+                }
+
+                _lockFileName = Path.Combine(rootDirectory, lockName);
+            }
+
+            public bool WaitSignal(int millisecondsTimeout)
+            {
+                DateTime maxWait = DateTime.UtcNow.AddMilliseconds(millisecondsTimeout);
+                try
+                {
+                    while (DateTime.UtcNow < maxWait)
+                    {
+                        using (new FileStream(_lockFileName, FileMode.OpenOrCreate, FileAccess.ReadWrite, FileShare.None)) { }
+                        Thread.Sleep(150);
+                    }
+                    return false;
+                }
+                catch (IOException)
+                {
+                    return true;
+                }
+            }
+
+            public bool TryAcquire(int millisecondsTimeout)
+            {
+                DateTime maxWait = DateTime.UtcNow.AddMilliseconds(millisecondsTimeout);
+                for (; ; )
```

Comment:
`while (true)` is more idiomatic c#.

Reply:
Both make the same code (or should - they do in latest Roslyn/RyuJIT). It is a matter of style. I only want to be consistent. In CoreFX I see 279 of the "for" format and 485 of the "while" format so I would use the latter.

Reply:
Sounds good Dan!I'll fix it ASAP. Thank's for your time!

Reply:
Fixed